python/res/enkf/jobs/enkf_scaling_get_data.py

```diff
@@ -0,0 +1,173 @@
+        obs_std = input_data.loc["STD_" + key]
+
+        filters.append(_filter_ensamble_std(ens_std, events.std_cutoff))
+        filters.append(_filter_ens_mean_obs(ens_mean, ens_std, obs_values, obs_std, events.alpha))
+
+    combined_filter = _combine_filters(filters)
+    return output_data.drop(columns=combined_filter[combined_filter].index)
+
+
+def _filter_on_column_index(df, index_list):
+    """
+    Takes a dataframe and retuns a subset where the columns in index_list are filtered out
+    """
+    df = deepcopy(df)
+    if isinstance(index_list, (list, tuple)):
+        if max(index_list) > df.shape[1]:
+            msg = ("Index list is larger than observation data, please check input, "
+                   "max index list: {} number of data points: {}".format)
+            raise IndexError(msg)
+        return df.iloc[:, list(index_list)]
+    else:
+        return df
+
+
+def _get_general_data(ert, observation_key):
+    ert_obs = ert.getObservations()
+    data_key = ert_obs[observation_key].getDataKey()
+
+    general_data = pd.DataFrame()
+
+    for time_step in range(ert_obs[observation_key].getActiveCount()):
+
+        data = GenDataCollector.loadGenData(ert, "default", data_key, time_step)
+
+        general_data = general_data.append(
+            pd.DataFrame(_get_observations(ert_obs, observation_key), index=["OBS_" + observation_key]))
+
+        general_data = general_data.append(
+            pd.DataFrame(_get_std(ert_obs, observation_key), index=["STD_" + observation_key]))
+
+        general_data = general_data.append(
+            pd.concat([pd.DataFrame([data[key]], index=[observation_key]) for key in data.keys()]))
+
+    return general_data
+
+
+def _get_summary_data(ert, observation_key):
+    return pd.concat([_add_summary_observations(ert, observation_key), _add_summary_data(ert, observation_key)])
+
+
+def _add_summary_data(ert, observation_key):
+    data = SummaryCollector.loadAllSummaryData(ert, "default", [observation_key])
+    data = data[observation_key].unstack(level=-1)
+    return data.set_index([[observation_key] * len(data)])
+
+
+def _add_summary_observations(ert, observation_key):
+    data = SummaryObservationCollector.loadObservationData(ert, "default", [observation_key]).transpose()
+    return data.set_index(data.index.str.replace(r"\b" + observation_key, "OBS_" + observation_key, regex=True))
+
+
+def _get_observations(all_obs, obs_key):
+    return [obs_node.get_data_points() for obs_node in all_obs[obs_key]]
+
+
+def _get_std(all_obs, obs_key):
+    return [obs_node.get_std() for obs_node in all_obs[obs_key]]
+
+
+def get_scaled_matrix(input_data, observation_keys):
+    """
+    Duplicates the behavior of obs_data_scale, and scales the simulated data
+    by 1 / (observation standard deviation), per observation key, i.e. each
+    simulation data point is scaled by its corresponding std deviation
+    from observations.
+    """
+    input_data = deepcopy(input_data)
```

Comment:
Redefining input parameters is risky and a bit unnecessary as you are not modifying the input data.

Reply:
:+1: